packages/trader/src/Modules/SmartChart/Components/all-markers.jsx

```diff
@@ -265,40 +328,77 @@ const TickContract = RawMarkerMaker(
+                bottom: barrier_2,
                 ctx,
+                fill_color: getColor({
+                    status: has_crossed_accu_barriers ? 'accu_shade_crossed' : 'accu_shade',
+                    is_dark_theme,
+                }),
+                labels: accu_barriers_difference,
+                previous_tick: {
+                    color: getColor({ status: 'fg', is_dark_theme }) + opacity,
+                    radius: 2 * scale,
+                },
                 start_left: start.left,
-                fill_color: 'rgba(55, 124, 252, 0.08)',
-                stroke_color: getColor({ status: 'dashed_border', is_dark_theme }),
+                stroke_color: getColor({ status: has_crossed_accu_barriers ? 'lost' : 'open', is_dark_theme }),
                 top: barrier,
-                bottom: barrier_2,
                 scale,
             });
             return;
         }
-
         if (
             barrier &&
             barrier_2 &&
-            ((previous_tick && is_accumulators_contract && is_in_contract_details) || (!contract_type && start))
+            (((previous_tick || exit) && is_accumulator_contract && is_in_contract_details) ||
+                (!contract_type && start))
         ) {
             // draw 2 barriers with a shade between them for an ongoing ACCU contract:
+            const is_accu_contract_open = isAccumulatorContractOpen(
+                { contract_type, current_spot: exit?.top, high_barrier: barrier, low_barrier: barrier_2, status },
+                true
+            );
+            const contract_details_start_left = is_accu_contract_open ? exit?.left : previous_tick?.left;
+            const preceeding_tick = ticks.length > 2 && is_accu_contract_open && ticks[ticks.length - 2];
             draw_shaded_barriers({
+                bottom: barrier_2,
                 ctx,
-                start_left: is_in_contract_details ? previous_tick.left : start.left,
-                fill_color: 'rgba(0, 167, 158, 0.08)',
+                fill_color: getColor({
+                    status:
+                        has_crossed_accu_barriers || status === 'lost' ? 'accu_shade_crossed' : 'accu_contract_shade',
+                    is_dark_theme,
+                }),
                 // we should show barrier lines in contract details even when they are outside of the chart:
                 has_persistent_borders: is_in_contract_details,
-                stroke_color: getColor({ status: 'dashed_border', is_dark_theme }),
-                top: barrier,
-                bottom: barrier_2,
+                labels: !is_in_contract_details && accu_barriers_difference,
+                previous_tick: is_in_contract_details
+                    ? {
+                          radius: is_mobile ? 1.5 : 2.5,
+                          stroke_color: getColor({ status: 'open', is_dark_theme }),
+                          preceeding_tick: {
+                              ...preceeding_tick,
```

Comment:
barriers in C.Details are now delayed, that's why to avoid overlapping, markers for previous tick and another tick, that precedes the previous tick when a contract is ongoing, are drawn in canvas.